src/main/kotlin/no/nav/syfo/behandler/BehandlerService.kt

```diff
@@ -33,4 +61,67 @@ class BehandlerService(
 
         return null
     }
+
+    private fun createOrGetBehandler(fastlege: Fastlege, personIdentNumber: PersonIdentNumber): Behandler {
+        val behandlerDialogmeldingForPersonList = database.getBehandlerDialogmeldingForPerson(
+            personIdentNumber = personIdentNumber,
+        )
+
+        val behandlerDialogmeldingForPersonsFastlege =
+            behandlerDialogmeldingForPersonList.find { (_, pBehandlerDialogmelding) -> pBehandlerDialogmelding.partnerId == fastlege.partnerId.toString() }
+
+        if (behandlerDialogmeldingForPersonsFastlege == null) {
+            return createBehandlerDialogmelding(
+                personIdentNumber = personIdentNumber,
+                newBehandler = fastlege.toNewBehandler()
+            )
+        }
+
+        val (arbeidstakerPersonident, pBehandlerDialogmelding) = behandlerDialogmeldingForPersonsFastlege
+        val isNyFastlege = behandlerDialogmeldingForPersonList.indexOf(behandlerDialogmeldingForPersonsFastlege) > 0
```

Comment:
Hm, skal det ikke være < 0 her? (dvs at behandleren ikke er blant de eksisterende behandlerene for personIdent'en?

Reply:
Tror det skal bli riktig at vi sjekker om behandleren er ulik siste lagret behandler. For å få historikk på bytte av fastlege (også bytte tilbake til en fastlege man har hatt tidligere). Kanskje tydeligere hvis jeg skriver om til:
`val isBytteAvFastlege = behandlerDialogmeldingForPersonList.first() != behandlerDialogmeldingForPersonsFastlege` ?

Reply:
Ah, ok så listen er ordnet slik at den nyeste har index=0? Ja, det blir kanskje enklere å forstå med forslaget ditt

Reply:
Refaktorert dette nå